### PR TITLE
8473 scrub does not detect errors on active spares

### PR DESCRIPTION
**Pull request by asomers**
ZFS should scrub spare devices that have already finished resilvering

### CODE REVIEW

**Comment by asomers** [comment]
Can I please get some review on this?  Without this change, ZFS will fail to detect corruption on an active spare, even when you scrub it.

**Comment by andy-js** [comment]
LGTM.

**Comment by asomers** [comment]
What else do we need to do for this PR?

**Comment by ahrens** [comment]
@asomers Can you tell us about any history of real-world use of this patch, e.g. in your product or in FreeBSD?

**Comment by asomers** [comment]
I discovered the problem using our version of the ZFS test suite.  I patched it in our product back in January 2013, but I have never pushed the patch upstream (to OpenZFS or FreeBSD) until now.

File: usr/src/uts/common/fs/zfs/vdev_mirror.c [hunk 1]
```diff
@@ -109,13 +112,51 @@ vdev_mirror_map_alloc(zio_t *zio)
...
+		 * If we are resilvering, then we should handle scrub reads
+		 * differently; we shouldn't issue them to the resilvering
+		 * device because it might not have those blocks.
+		 *
+		 * We are resilvering iff:
+		 * 1) We are a replacing vdev (ie our name is "replacing-1" or
+		 *    "spare-1" or something like that), and
+		 * 2) The pool is currently being resilvered.
+		 *
+		 * We cannot simply check vd->vdev_resilvering, because that
```
[the remark this answers is not in the document]
Comment (asomers) [reply usr/src/uts/common/fs/zfs/vdev_mirror.c]: I guess a lot changes in 3.5 years.  I just did a little experiment, and the replacement `vdev_resilver_txg` is also not set in this case.  I'll update the comment.

**Comment by asomers** [comment]
Dear reviewers: my most recent commit is just a comment change, so no retest is necessary.

**Pull request closed without merging**